test/Adapter/PhpCodeTest.php

```diff
@@ -31,106 +31,117 @@ public function tearDown()
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-    public function testSerializeNumeric()
-    {
-        $value    = 100.12345;
-        $expected = '100.12345';
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
     public function testSerializeObject()
     {
-        $value    = new \stdClass();
-        $expected = "stdClass::__set_state(array(\n))";
+        $object = new Dummy();
+        $expected = Encoder::encode($object);
 
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-    public function testUnserializeString()
-    {
-        $value    = "'test'";
-        $expected = 'test';
+        $data = $this->adapter->serialize($object);
 
-        $data = $this->adapter->unserialize($value);
+        $this->assertNotEmpty($data);
         $this->assertEquals($expected, $data);
     }
 
-    public function testUnserializeFalse()
+    public function testUnserializeObject()
     {
-        $value    = 'false';
-        $expected = false;
+        $expected = new Dummy();
 
-        $data = $this->adapter->unserialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-    public function testUnserializeNull()
-    {
-        $value    = 'NULL';
-        $expected = null;
+        $serialized = Encoder::encode($expected);
+        $data = $this->adapter->unserialize($serialized);
 
-        $data = $this->adapter->unserialize($value);
-        $this->assertEquals($expected, $data);
+        $this->assertInstanceOf(get_class($expected), $data);
     }
 
-    public function testUnserializeNumeric()
-    {
-        $value    = '100';
-        $expected = 100;
-
-        $data = $this->adapter->unserialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-/* TODO: PHP Fatal error:  Call to undefined method stdClass::__set_state()
-    public function testUnserializeObject()
+    /**
+     * @dataProvider dataProvider
+     */
+    public function testPhpCode($value, $expected, $serialize)
     {
-        $value    = "stdClass::__set_state(array(\n))";
-        $expected = new stdClass();
+        if ($serialize) {
+            $data = $this->adapter->serialize($value);
+        } else {
+            $data = $this->adapter->unserialize($value);
+        }
 
-        $data = $this->adapter->unserialize($value);
         $this->assertEquals($expected, $data);
     }
-*/
 
-    public function testUnserializeInvalid()
+    public function dataProvider()
     {
-        if (version_compare(PHP_VERSION, '7', 'ge')) {
-            $this->markTestSkipped('Cannot catch parse errors in PHP 7+');
-        }
-        $value = 'not a serialized string';
-
-        $this->setExpectedException('Zend\Serializer\Exception\RuntimeException', 'syntax error');
-        $this->adapter->unserialize($value);
+        return [
+            [
```

Comment:
I think this test is more legible if formated as columns

Ex: https://github.com/zendframework/zend-inputfilter/blob/a05fa0bfca3a82ef119e1af2e970a48d787913c7/test/InputTest.php#L572-L589